tests/ui_tools/test_popups.py

```diff
@@ -1604,6 +1605,51 @@ def test_checkbox_toggle_visual_notification(
         toggle_visual_notify_status.assert_called_once_with(stream_id)
 
 
+class TestTopicInfoView:
+    @pytest.fixture(autouse=True)
+    def mock_external_classes(
+        self, mocker: MockerFixture, general_stream: Dict[str, Any], topics: List[str]
+    ) -> None:
+        self.controller = mocker.Mock()
+
+        mocker.patch.object(
+            self.controller, "maximum_popup_dimensions", return_value=(64, 64)
+        )
+        mocker.patch(LISTWALKER, return_value=[])
+        self.stream_id = general_stream["stream_id"]
+        self.topic = topics[0]
+        self.controller.model.stream_dict = {self.stream_id: general_stream}
+
+        self.topic_info_view = TopicInfoView(
+            self.controller, self.stream_id, self.topic
+        )
+
+    @pytest.mark.parametrize(
+        "topic_name, expected_topic_button_label",
+        [
+            ("hi!", "Resolve Topic"),
+            (f"{RESOLVED_TOPIC_PREFIX}" + "hi!", "Unresolve Topic"),
+        ],
+    )
+    def test_topic_button_label(
+        self, topic_name: str, expected_topic_button_label: str
+    ) -> None:
+        topic_info_view = TopicInfoView(self.controller, self.stream_id, topic_name)
+        assert (
+            topic_info_view.resolve_topic_setting_btn_lbl == expected_topic_button_label
+        )
+
+    def test_toggle_resolve_status(self) -> None:
+        resolve_button = self.topic_info_view.widgets[-1]
+        resolve_button._emit("click")
+
+        self.controller.model.toggle_topic_resolve_status.assert_called_once_with(
+            stream_id=self.stream_id, topic_name=self.topic
+        )
+
+        self.controller.exit_popup.assert_called_once()
```

Comment:
This is fine, though I'll leave this comment unresolved since it may be useful to take this to a discussion of how best to integrate urwid and our own key handling further in this way.